index.html

```diff
@@ -12,32 +12,20 @@
 <body>
   <div class="container">
     <div class="row">
-      <div class="col-lg-6 col-md-12">
-        <!-- ko foreach: toDoLists -->
-        <div class="row toDoList">
-          <!-- ko component: 'toDoList' -->
-          <!-- /ko -->
-        </div>
-        <button class="btn btn-block btn-danger" data-bind="click: $parent.removeToDoList">Remove</button>
+      <!-- ko foreach: widgets -->
+      <div class="col-lg-3 col-md-4 col-sm-6 col-xs-12">
+        <!-- ko component: name -->

```

Comment:
We tell knockout to render the widget specified by the name property of the widget. See the app.js file, where we are setting the name property to either `'calculator'` or `'todo-list'` depending on what the user wants to add.